nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -500,6 +523,30 @@ def add_electrical_series(
     Missing keys in an element of metadata['Ecephys']['ElectrodeGroup'] will be auto-populated with defaults
     whenever possible.
     """
+    if isinstance(recording, RecordingExtractor):
+        # If the object is a SubRecording, we recover the gains either the first parent with non-default gains, or the
+        # highest level parent.
+        channel_conversion = recording.get_channel_gains()
+        default_channel_conversion = np.ones_like(channel_conversion)
+        channel_offset = recording.get_channel_offsets()
+        temp_recording = recording
+        while isinstance(temp_recording, SubRecordingExtractor):
+            parent_recording = temp_recording._parent_recording
+            channel_conversion = parent_recording.get_channel_gains()
+            channel_offset = parent_recording.get_channel_offsets()
+
+            default_channel_conversion = np.ones_like(channel_conversion)
+            if np.any(channel_conversion != default_channel_conversion):
+                break
+            else:
+                temp_recording = parent_recording
+        checked_recording = OldToNewRecording(oldapi_recording_extractor=recording)
+        if np.any(channel_conversion != default_channel_conversion):
+            checked_recording.set_channel_gains(gains=channel_conversion)
+        if np.any(channel_offset != np.zeros_like(channel_offset)):
+            checked_recording.set_channel_offsets(offsets=channel_offset)
```

Comment:
@CodyCBakerPhD this could also happen when the recording is a `BasePreprocessorRecordingExtractor`. I woudl suggest to merge as is and fix it later. Do you agree?

Reply:
Very interesting - good to know! Fine to delay that since we've never made use of that specific type.

Reply:
@alejoe91 If you think the logic applies generally enough, sure! I know it could be interpreted as a problem on other non-NWB RecordingExtractor contexts of translating an old to a new.